pkg/apis/kubermatic/v1/datacenter.go

```diff
@@ -458,6 +458,12 @@ type DatacenterSpecNutanix struct {
 	Endpoint string `json:"endpoint"`
 	// Optional: Port to use when connecting to the Nutanix Prism Central endpoint (defaults to 9440)
 	Port *int32 `json:"port,omitempty"`
+
+	// PeEndpoint to use for accessing Nutanix Prism Element. No protocol or port should be passed,
+	PeEndpoint string `json:"peEndpoint"`
+	// Optional: PePort to use when connecting to the Nutanix Prism Element endpoint (defaults to 9440)
+	PePort *int32 `json:"pePort,omitempty"`
```

Comment:
I'm still thinking about this, but I believe this should be in the Preset / Cluster, not the Seed. Here's my reasoning:

Currently, we treat one Prism Central instance as a KKP datacenter. One Prism Central can manage multiple NTNX clusters. Our `Cluster` resources have the `NutanixCloudSpec.ClusterName` key, which means you can select the NTNX cluster you'd like your user cluster running on _during user cluster creation_.

But Prism Element is the API for one NTNX cluster. Defining this at the `Seed` level means that you already lock everything to a single NTNX cluster, as the CSI driver will always talk to that NTNX cluster. Deploying to a different NTNX cluster means this breaks in bad ways.

IMO we have four ways out of this:

1. Lock one KKP Nutanix datacenter to a specific NTNX cluster managed by the target Prism Central. I'm not a fan of this, because it might create a lot of repetition if you manage several NTNX clusters from a Prism Central. Defeats the purpose of that to a certain degree IMHO.
2. Change the `Seed` datacenter spec to contain a `map[string]PrismElementConfig{}` with `PrismElementConfig` having an `endpoint` and a `port` key. That way, you can define multiple PE endpoints for NTNX cluster names at the Seed level.
3. See if we can be smart and query this information from Prism Central for a certain NTNX cluster name. That would probably be awesome, but not sure that's going to work.
4. Move this to the `Cluster` spec and set it in a `Preset` together with username and password.

WDYT, @WeirdMachine?

Reply:
Yes, lets move it to the cluster spec. This is also a requirement to query it from the nutanix api.